### DISCUSSION TITLE
关于控制台应用用终端显示

### DISCUSSION THREAD

**Discussion by weishengovo** [BYJRK/DotNet-Discussions#83] [通用]
up主 控制台应用程序跑起来是终端显示
![image](https://user-images.githubusercontent.com/71328027/232356338-911797ce-4db6-4ed4-8fe2-2ec7bdda9e8c.png)
我自己 控制台应用程序跑起来是cmd显示
![image](https://user-images.githubusercontent.com/71328027/232356466-b67597e7-6044-4bb1-bf4b-47ad26246c30.png)
是需要那里设置的吗

**Comment by BYJRK** [comment]
说实话我不知道🤣自从用了win11，直接就是这样，啥都没调。我估计可能需要配置全局默认cmd打开方式为终端，或者vs里面也有类似的设置？

**Comment by KiyanYang** [comment]
在 Windows Terminal 里面设置默认终端为 Windows Terminal。

![Snipaste_2023-04-17_13-06-18](https://user-images.githubusercontent.com/50903774/232383787-a3703a1b-0e83-46ea-b648-0ef0b374e6c7.png)

**Reply by weishengovo** [reply]
我设置成这个还是一样的
![image](https://user-images.githubusercontent.com/71328027/232476950-af95d6a8-a059-45d3-b2e0-de0be2173b04.png)

**Reply by KiyanYang** [reply]
哈哈，那我就不知道了，我只调了这个就好了🤣

**Comment by Dynesshely** [comment]
在 Win11 中调整以下设置即可

![image](https://user-images.githubusercontent.com/50760269/233887008-fb055f8a-fbfa-4d93-9658-07176471c0bf.png)

**Reply by weishengovo** [reply]
![image](https://user-images.githubusercontent.com/71328027/233903509-0a19b06e-9bb8-48a5-b565-1392a9e702af.png)
你好 这里设置还是一样的

**Reply by Dynesshely** [reply]
那就不是很清楚是哪里设置的问题了
也许 VS 某个设置可以调整
我爱莫能助啦

**Comment by weishengovo** [comment]
我在gitHub中找到了这个问题
https://github.com/microsoft/terminal/issues/13047
最后一个回答解决了我的问题 只要visual studio不用管理员 运行就可以用终端显示了 ？？？ 这是什么奇怪的bug

**Reply by Dynesshely** [reply]
似乎不是 Visual Studio 的问题, 应该是 Terminal 的问题

我的猜测如下:
当 Windows 默认控制台主机设置为 Terminal 之后, Terminal 应该会尝试拦截控制台程序的启动, 然后通过 Terminal 的界面来提供交互
可能 Terminal 的拦截服务或是其它什么东西不能拦截管理员权限运行的控制台程序, 或是没有做相关判断
导致管理员权限运行的控制台程序仍然会使用默认终端执行
我在本地尝试直接管理员权限运行控制台程序的 exe 文件, 也不能自动使用 Terminal 的界面, 而是默认控制台主机

可以确定的是,
当 Visual Studio 运行在管理员权限下时, 使用 Visual Studio 调试应用也将以管理员权限调试应用

**Reply by Dynesshely** [reply]
我刚才怀疑是否是 dotnet 的问题, 因为 .net core 以来, 生成的 exe 只是一个调用 dotnet 命令的包装器
因此我使用 C++ 编写的控制台程序进行了确认
可以肯定不是 dotnet 的问题
那么只能是 Terminal 的问题了
